### DISCUSSION TITLE
Status of the “annotations.reader” service?

### DISCUSSION THREAD

**Discussion by alexisLefebvre** [symfony/symfony#45304] [Q&A] [answered]
Hello,
I'm wondering if the “annotations.reader” service is supported? It is never mentioned in the official documentation.

## Context

The CI of LiipTestFixturesBundle was suddenly [broken](https://github.com/liip/LiipTestFixturesBundle/runs/4776332809?check_suite_focus=true) with the following message:

> The service "Liip\TestFixturesBundle\Services\DatabaseToolCollection" has a dependency on a non-existent service "annotations.reader".

I was able to avoid the error by requiring an older version of `doctrine/persistence`: https://github.com/liip/LiipTestFixturesBundle/pull/169/files

The only explanation I got is that [this change](https://github.com/doctrine/persistence/pull/197/files) resulted in the lack of the `Doctrine\Common\Annotations\AnnotationReader` class.

This service is declared [here](https://github.com/symfony/symfony/blob/a061fc4a3db54ceaf3b3fd821d35beb047d967f1/src/Symfony/Bundle/FrameworkBundle/Resources/config/annotations.php#L26), yet it was unclear why the service may be missing.

@derrabus [told me](https://github.com/doctrine/persistence/pull/197#issuecomment-1025916246):

> If the library is missing, Symfony won't create that service.

So it appears than this service can be undefined, despite the fact that it is in the FrameworkBundle.

## Proposals

- explain “annotations.reader” in the documentation
- explain that `doctrine/annotations` is required instead of only showing a vague error “non-existent service "annotations.reader"”.

**Comment by derrabus** [comment, marked as answer]
Please don't hesitate to submit a PR to the docs repository!

That being said, if LiipTestFixturesBundle uses Doctrine Annotations, it should either require `doctrine/annotations` as a mandatory dependency or contain fallback logic for the case that the library is not present.

**Reply by alexisLefebvre** [reply]
Yes that's what we did: https://github.com/liip/LiipTestFixturesBundle/pull/170/files

**Reply by alexisLefebvre** [reply]
Here is the PR: https://github.com/symfony/symfony-docs/pull/16511
